src/Appwrite/Promises/Promise.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Appwrite\Promises;
```

Comment:
This should not belong in Appwrite. We should consider whether it should be a utopia library and avoid it for now. How much do we rely on this class?

Reply:
We need this to allow async query resolution. I've followed the pattern wherein the whole `Promises` directory could be it's own utopia library when we want to abstract it from Appwrite

Reply:
Let's add it as a task in the backlog